ruby_on_rails/advanced_forms_and_activerecord/installing_psql.md

```diff
@@ -0,0 +1,342 @@
+
+#### 3.3 Creating the Role Database
+One other important step in setting up Postgre is that each role must have its own database of the same name. Without it, the role we just created will not be able to log in or interact with the Postgre server.
+
+You can try to run `psql` now, but you will get an error that the database does not exist. Not to worry, let's create one to resolve fix this:
+
+~~~bash
+sudo -i -u postgres createdb <linux_username>
+~~~
+
+Now our role is fully set up: we've got `<role_name>` and that role has a database.
+
+#### 3.4 Securing Our New Role
+One important thing that we have to do is to set up a password for our new role so that the data is protected. Now that our role is set up, we can actually use it to administer the Postgre server. All you have to do is enter this command to get into the PostgreSQL prompt:
+
+~~~bash
+psql
+~~~
+
+You should see the PostgreSQL prompt come up with the new role we just created, like so:
+
+~~~
+<role_name>=#
+~~~
+
+If you don't see a similar prompt, then reach out on [Discord](https://discord.gg/V75WSQG) for some help. 
+
+If you **do** see a similar prompt, then we can create a password for the role like so:
+
+~~~
+\password <role_name>
+~~~
+
+You'll be prompted to enter a password and to verify it. Once you are done, the prompt will return to normal. Now, we will configure the permissions for our new role:
+
+~~~
+grant all privileges on database <role_database_name> to <role_name>;
+~~~
+
+Remember that you should change the `<role_database_name>` and `<role_name>` (they should both the same)! If you see `GRANT` in response to the command, then you can type `\q` to exit the prompt.
+
+This role and password combination enables us to log in to the PostgreSQL service to perform operations, so let's save this into the environment so local Rails apps can do those things.
+
+#### 3.5 Saving Access Information in the Environment
+After finishing our configuration, the last step is save it into the environment to access later.
+
+In order to save our password to the environment, we can run this command:
+
+~~~bash
+echo 'export DATABASE_PASSWORD="<role_password>"' >> ~/.bashrc
+~~~
+
+Note here the name we've chosen for our environment variable: `DATABASE_PASSWORD`. Also, remember to update `<role_password>` in the command to what was set above!
+
+Now, this variable lives in our environment for us to use. As the variable is new, we'll want to reload the environment so that we can access it. To reload the environment, you can close and re-open your terminal.
+
+Once that's done, we can move to testing it out!
+
+</details>
+
+<details markdown="block">
+<summary class="dropDown-header">macOS
+</summary>
+
+### Step 1: Make sure the system is up to date
+Before running commands with homebrew, you'll want to make sure things are up to date. Run the following commands.
```

Comment:
Add a colon in line with other cases like this also add note to run the command sequentially. Sometimes people copy-paste the entire thing instead.

```suggestion
Before running commands with homebrew, you'll want to make sure things are up to date. Run the following commands one by one:
```